src/features/island/buildings/components/building/Building.tsx

```diff
@@ -164,20 +176,23 @@ const DestroyedBuilding: React.FC<Prop> = ({
       <Modal show={showModal} onHide={() => setShowModal(false)}>
         <CloseButtonPanel onClose={() => setShowModal(false)}>
           <div className="p-2">
-            <Label icon={tornadoIcon} type="danger" className="mb-1 -ml-1">
-              {t("tornado")}
+            <Label
+              icon={DESTROYED_BUILDING_ICONS[calendarEvent]}
+              type="danger"
+              className="mb-1 -ml-1"
+            >
+              {t(calendarEvent)}
             </Label>
             <p className="text-sm">
-              {t("tornado.building.destroyed.description")}
+              {t(`${calendarEvent}.building.destroyed.description`)}
             </p>
             <Label
               icon={SUNNYSIDE.icons.stopwatch}
               type="transparent"
               className="mt-2 ml-1"
             >
               {`Ready in: ${secondsToString(
-                24 * 60 * 60 -
-                  (Date.now() - game.calendar.tornado!.triggeredAt) / 1000,
+                (Date.now() - game.calendar[calendarEvent]!.triggeredAt) / 1000,
```

Comment:
This time appears to be going backwards. It keeps increasing 😅 

<img width="550" alt="Screenshot 2025-01-14 at 9 25 39 AM" src="https://github.com/user-attachments/assets/1369aaa2-cc1a-491c-8846-1e7e659c75ee" />

Reply:
Nice catch! Fixed